Alloy/MTLCommandQueue+Extensions.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  MTLCommandQueue+Extensions.swift
+//  Alloy
+//
+//  Created by Eugene Bokhan on 27.11.2019.
+//
+
+import Metal
+
+public extension MTLCommandQueue {
+
+    func scheduleAndWait<T>(_ bufferEncodings: (MTLCommandBuffer) throws -> T) throws -> T {
+        guard let commandBuffer = self.makeCommandBuffer()
+        else { throw CommonErrors.metalInitializationFailed }
```

Comment:
I think in this particular case dedicated error was actually fine. why? because there are situation when failed buffer creation is expected - i.e. for queue with limited capacity.